mlflow/telemetry/track.py

```diff
@@ -61,25 +68,23 @@ def _generate_telemetry_record(
         if params and params[0] == "cls" and isinstance(arguments["cls"], type):
             del arguments["cls"]
 
-        params_class = PARAMS_MAPPING.get(func.__name__)
-        record_params = params_class.parse(arguments) if params_class else None
-        return APIRecord(
-            api_module=func.__module__,
-            api_name=func.__qualname__,
+        record_params = event.parse(arguments)
+        return Record(
+            event_name=event.name,
             timestamp_ns=time.time_ns(),
             params=record_params,
-            status=APIStatus.SUCCESS if success else APIStatus.FAILURE,
+            status=Status.SUCCESS if success else Status.FAILURE,
             duration_ms=duration_ms,
         )
     except Exception:
         return
 
 
-def _is_telemetry_disabled_for_func(func: Callable[..., Any]) -> bool:
+def _is_telemetry_disabled_for_event(event: type[Event]) -> bool:
     try:
         if client := get_telemetry_client():
             if client.config:
-                return func.__qualname__ in client.config.disable_api_map.get(func.__module__, [])
+                return event.name in client.config.disable_events
             # config is not fetched yet, so we assume telemetry is enabled
```

Comment:
If the config is not fetched yet, should we disable telemetry?

Reply:
Yes, I'll update it in the next PR :)